specification/sbi_atee.adoc

```diff
@@ -1093,6 +1093,190 @@ The possible error codes returned in `sbiret.error` are shown below.
+|===
+
+[#sbi_tee_interrupt_unbind_tvm_cpu_aia_imsic_end]
+=== Function: TEE Interrupt Bind TVM CPU AIA IMSIC End (FID #6)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_unbind_tvm_cpu_aia_imsic_end(unsigned long tvm_guest_id,
+                                                             unsigned long tvm_vcpu_id);
+-------
+
+Completes the unbinding process for the specified vCPU from its guest interrupt files after
+a TLB flush sequence for the TVM has been completed. The interrupt state is saved to the vCPU's
+software interrupt file and the guest interrupt files are free to be reclaimed via
+`sbi_tee_interrupt_reclaim_tvm_aia_imsic()` or bound to another vCPU via `sbi_tee_interrupt_unbind_tvm_cpu_aia_imsic_begin()`.
+Must be called on the physical CPU to which the vCPU is bound. Upon success, the vCPU is 
+free to be bound to another physical CPU.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_unbind_tvm_cpu_aia_imsic_end_errors]
+.TEE Interrupt Unbind TVM CPU AIA IMSIC End
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation was completed successfully.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `tvm_vcpu_id` were invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+[#sbi_tee_interrupt_inject_tvm_cpu]
+=== Function: TEE Interrupt Inject TVM CPU (FID #7)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_inject_tvm_cpu(unsigned long tvm_guest_id,
+                                               unsigned long tvm_vcpu_id
+                                               unsigned long interrupt_id);
+-------
+
+Injects an external interrupt with the given `interrupt_id` into the specified vCPU. If the
+vCPU is presently bound to an IMSIC guest interrupt file, the interrupt is immediately
+injected by writing to the interrupt file. If it is not bound, the interrupt is recorded
+in the software and will be injected once the vCPU becomes bound. The specified interrupt ID
+must be valid and must have been allowed by the guest with `sbi_tee_guest_allow_external_interrupt()`.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_interrupt_inject_tvm_cpu_errors]
+.TEE Interrupt Inject TVM CPU
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `tvm_vcpu_id` or `interrupt_id` were invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+[#sbi_tee_interrupt_rebind_tvm_cpu_aia_imsic_begin]
+=== Function: TEE Interrupt Rebind TVM CPU AIA IMSIC Begin (FID #8)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_rebind_tvm_cpu_aia_imsic_begin(unsigned long tvm_guest_id,
```

Comment:
that function name is quite the mouthful

Reply:
I have removed the `tvm_cpu` part as it's quite obvious from the arguments and also the way imsic files work.